sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareSetMetadataOptions.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.share.options;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.storage.file.share.models.ShareRequestConditions;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Extended options that may be passed when setting metadata on a share.
+ */
+@Fluent
+public class ShareSetMetadataOptions {
+
+    private Map<String, String> metadata;
+    private ShareRequestConditions requestConditions;
+
+    /**
+     * @return Metadata to set on the share, if null is passed the metadata for the share is cleared.
+     */
+    public Map<String, String> getMetadata() {
+        return metadata == null ? null : Collections.unmodifiableMap(metadata);
```

Comment:
Do we need to make it an unmodifiableMap in both the setter and getter?

Reply:
I think yes cause otherwise a user could get the map and then modify it and theyd be modifying the map we care about